### PR TITLE
feat(snap): add go-based configure and install hooks

### PR DESCRIPTION
**Pull request by siggiskulason**
This commit adds go-based configure and install hooks to the
snap which adds additional support for environment variable
configuration overrides (see snap/README.md for details).

This commit also adds a content interface plug to allow
configuration.toml, devices, and profiles to be provided
by an external snap.

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-modbus-go/blob/master/.github/Contributing.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## New Imports

- [ ] Yes
- [ ] No

### CODE REVIEW

File: snap/local/hooks/cmd/install/install.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+	err = installConfig()
+	if err != nil {
+		hooks.Error(fmt.Sprintf("edgex-device-modbus:install: %v", err))
+		os.Exit(1)
+	}
+
+	// disable the service and handle the autostart logic in the configure hook
+	// as default snap configuration is not available when the install hook runs
+	err = cli.Stop("device-modbus", true)
```
Comment (tonyespy) [inline snap/local/hooks/cmd/install/install.go:89]: We replaced this in app-service-configurable and device-camera by using disabling the service directly in `snapcraft.yaml`:

```
    passthrough:
      install-mode: disable
```

File: snap/README.md [hunk 2]
```diff
@@ -32,3 +69,99 @@ ensures that as well as starting the service now, it will be automatically start
...
+
+For details on the mapping of configuration options to Config options, please refer to "Service Environment Configuration Overrides".
+
+## Service Environment Configuration Overrides
+
+**Note** - all of the configuration options below must be specified with the prefix: 'env.'
+
+```
+[Service]
+service.boot-timeout            // Service.BootTimeout
```
Comment (tonyespy) [inline snap/README.md:152]: These are the old values, I think you should just be able to cut & paste this section from device-camera....

**Review by tonyespy — CHANGES_REQUESTED** [review]
Two comments inline...

**Review by tonyespy — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**